File: ClientApp/src/app/models/ProjectSummary.ts
import { ProjectSummaryMonth } from "./ProjecSummaryMonth";

export class ProjectSummary {
    DailyWorkHours = 0;
    PerceGlobalProgress = 0;
    TotalHoursPlanned = 0;
    TotalProgress = 0;    
    TotalTasks = 0;
    TotalTasksThisMonth = 0;
    TotalHoursThisMonth = 0;
    TotalProgressThisMonth = 0; 
    PerceProgressThisMonth = 0;  
    TotalHoursNextMonth = 0;
    TotalTasksNextMonth = 0;
    TotalProgressNextMonth = 0;
    PerceProgressNextMonth = 0;
    WeeklyWorkHours = 0;
    MonthlyWorkHours = 0;
    PercePlannedHoursThisMonth = 0;
    PercePlannedHoursNextMonth = 0;
    PercePlannedRange = 0;
    ProjectSummaryMonths : ProjectSummaryMonth[] = [];
}

File: model/ProjectSummary.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace OCPlanner
{
    public class ProjectSummary
    {
        public decimal DailyWorkHours;
        public decimal WeeklyWorkHours;
        public decimal MonthlyWorkHours;
        public int PerceGlobalProgress;
        public int TotalProgress;
        public int TotalHoursPlanned = 0;
        public int TotalTasks = 0;
        public int TotalHoursThisMonth = 0;
        public int TotalTasksThisMonth = 0;
        public int TotalProgressThisMonth = 0;
        public int PerceProgressThisMonth = 0;
        public int TotalHoursNextMonth = 0;
        public int TotalTasksNextMonth = 0;
        public int TotalProgressNextMonth = 0;
        public int PerceProgressNextMonth = 0;
        public int PercePlannedHoursThisMonth = 0;
        public int PercePlannedHoursNextMonth = 0;
        public int PercePlannedRange = 0;
        public List<ProjectSummaryMonth> ProjectSummaryMonths = new List<ProjectSummaryMonth>();


    }
}
